cunumeric/deferred.py

```diff
@@ -1132,6 +1132,40 @@ def dot(self, src1, src2, stacklevel=0, callsite=None):
                 f"dot between {rhs1_array.ndim}d and {rhs2_array.ndim}d arrays"
             )
 
+    # Create array from input array and indices
+    @profile
+    @auto_convert([1])
+    @shadow_debug("choose", [1])
+    def choose(
```

Comment:
The convention has been that the `self` argument on thunk methods is the array that gets filled in. So it should be `a` that gets passed as an argument, and `out` should become the `self` argument for this method.